core/src/main/java/jenkins/security/s2m/FilePathRuleConfig.java

```diff
@@ -40,7 +40,7 @@ protected FilePathRule parse(String line) {
         if (line.isEmpty())     return null;
 
         line = line.replace("<BUILDDIR>","<JOBDIR>/builds/<BUILDID>");
-        line = line.replace("<BUILDID>","[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]_[0-9][0-9]-[0-9][0-9]-[0-9][0-9]");
+        line = line.replace("<BUILDID>","(?:[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9]_[0-9][0-9]-[0-9][0-9]-[0-9][0-9]|[0-9]+)");
```

Comment:
> I have no idea what this does

`(?:…)` syntax? Non-binding group. (High-precedence grouping operator like `(…)` but without consuming a `Matcher.group` number.)

> hasn't the migration changed the buildIDs by now and removed the old timestamped folders from disk?

It should have, and this could probably be replaced with simple `"[0-9]+"`; I made the more conservative change for now. Can switch it if you think there is some risk to allowing writes into legacy build ID directories (I cannot think of any).

> is buildID 0 valid

No, but permitting a plugin to write into `<BUILDDIR>/0/somefile` is harmless.

> what about looping

I got 2³¹ problems but rejected callables ain’t one!